src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -821,7 +834,16 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
     # Retrieving Helm chart OCI Artifact location
-    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+    if container_registry_repository:
+        if auto_upgrade != "false":
```

Comment:
You would also have to check the existing helm values, if the auto-update is enabled.

Reply:
Yes agree. Its possible that the auto update is already set to false on the existing cluster, in that case we should not throw error

Reply:
ack